Source/WebCore/editing/TextIterator.h

```diff
@@ -128,29 +129,29 @@ class TextIterator {
     const TextIteratorBehaviors m_behaviors;
 
     // Current position, not necessarily of the text being returned, but position as we walk through the DOM tree.
-    Node* m_node { nullptr };
+    WeakPtr<Node, WeakPtrImplWithEventTargetData> m_currentNode;
```

Comment:
In general, we replace Raw pointers with WeakPtr or CheckedPtr. Node currently subclasses CanMakeWeakPtr but not CanMakeCheckedPtr. It is unclear if it is OK to make Node larger by subclassing both but we probably should try.

Now you're suggesting `RefPtr`. This might be possible but it is risky in general as we might introduce a reference cycle. It has to be done very carefully.

Reply:
I talked to @rniwa and he thinks we should try making Node subclass CanMakeCheckedPtr.

Reply:
I will look into having Node subclass CanMakeCheckedPtr in a follow-up.

Reply:
Text iterator is a case that won’t introduce cycles. It’s a worker object not owned by anything; it’s basically always on thes tack in a local. It should use `RefPtr`, not `WeakPtr`. Elsewhere I understand why `RefPtr` could be risky.

Reply:
Ok, I'll make the change. Thanks for the clarification.